src/test/java/com/cdancy/bitbucket/rest/features/SupportApiMockTest.java

```diff
@@ -0,0 +1,89 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.features;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import com.cdancy.bitbucket.rest.BaseBitbucketMockTest;
+import com.cdancy.bitbucket.rest.domain.support.SupportZip;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Mock tests for the {@link SupportApi} class.
+ */
+@Test(groups = "unit", testName = "SupportApiMockTest")
+public class SupportApiMockTest extends BaseBitbucketMockTest {
+
+    private String taskIdRegex = "^\\w+\\-+\\w+\\-+\\w+\\-+\\w+\\-+\\w+$";
+    private String supportZipRestApiPath = "/rest/troubleshooting/latest/support-zip" ;
+
+    public void testSupportZipCreation() throws Exception {
+        final MockWebServer mockWebServer = mockWebServer();
+        mockWebServer.enqueue(new MockResponse().setBody(payloadFromResource("/support-zip-create.json")).setResponseCode(200));
+        try {
+            final SupportApi supportApi = api(mockWebServer.getUrl("/")).supportApi() ;
+            final SupportZip supportZip = supportApi.createSupportZip();
+
+            assertSupportZipCreateRequest(mockWebServer);
+            assertIntegrity(supportZip);
+        } finally {
+            mockWebServer.shutdown();
+        }
+    }
+
+    public void testSupportZipStatus() throws Exception {
+        final MockWebServer mockWebServer = mockWebServer();
+        mockWebServer.enqueue(new MockResponse().setBody(payloadFromResource("/support-zip-create.json")).setResponseCode(200));
+        mockWebServer.enqueue(new MockResponse().setBody(payloadFromResource("/support-zip-request-status.json")).setResponseCode(200));
+        try {
+            final SupportApi supportApi = api(mockWebServer.getUrl("/")).supportApi() ;
+            SupportZip supportZip = supportApi.createSupportZip();
+            supportZip = supportApi.getSupportZipStatus(supportZip.taskId());
+
+            assertSupportZipCreateRequest(mockWebServer);
+            assertIntegritySupportZipStatus(supportZip);
+            assertSupportZipStatusRequest(mockWebServer, supportZip.taskId());
+        } finally {
+            mockWebServer.shutdown();
+        }
+    }
```

Comment:
We also need mock tests for when things fail on both endpoints. To trigger the failure you could use a non-existent taskId for one, combined with an error code of whatever that is supposed to return, and the other I would use an error code response of say `500` to account for insufficient privileges to do this task.